### PR TITLE
kfam backend

### PR DESCRIPTION
**Pull request by kunmingg**
kubeflow access management backend

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3386)

### CODE REVIEW

**Comment by kunmingg** [comment]
@kkasravi 
Thanks! Cleaned up duplicated code and auto-generated code.

**Comment by kunmingg** [comment]
@lluunn

File: components/access-management/api/swagger.yaml [hunk 1]
```diff
@@ -154,27 +172,34 @@ definitions:
...
     type: "object"
-    description: "in consistent with io.k8s.api.rbac.v1.Subject"
     properties:
       kind:
         type: "string"
       name:
         type: "string"
       namespace:
         type: "string"
+    description: "in consistent with io.k8s.api.rbac.v1.Subject"
```
Comment (avdaredevil) [inline components/access-management/api/swagger.yaml]:
```suggestion
    description: "Consistent with io.k8s.api.rbac.v1.Subject"
```

File: components/access-management/api/swagger.yaml [hunk 2]
```diff
@@ -154,27 +172,34 @@ definitions:
...
     type: "object"
-    description: "in consistent with io.k8s.api.rbac.v1.RoleRef"
     properties:
       apiGroup:
         type: "string"
       kind:
         type: "string"
       name:
         type: "string"
+    description: "in consistent with io.k8s.api.rbac.v1.RoleRef"
```
Comment (avdaredevil) [inline components/access-management/api/swagger.yaml]:
```suggestion
    description: "Consistent with io.k8s.api.rbac.v1.RoleRef"
```

File: components/access-management/api/swagger.yaml [hunk 3]
```diff
@@ -58,83 +61,98 @@ paths:
         500:
           description: "Internal Server Error"
   /v1/bindings:
-    post:
-      summary: "Create a binding specifying a user and an namespace, the binding will grant user edit-access to referred namespace"
-      description: "Only admin or referred namespace owner can create binding"
-      operationId: "create binding"
+    get:
+      summary: "Get bindings created via kubeflow; filtered by query param"
+      description: "Bindings will contain its current status"
```
Comment (avdaredevil) [inline components/access-management/api/swagger.yaml]:
```suggestion
      description: "Bindings will contain their current status"
```

File: components/access-management/api/swagger.yaml [hunk 4]
```diff
@@ -58,83 +61,98 @@ paths:
...
         500:
           description: "Internal Server Error"
-    get:
-      summary: "Get bindings created via kubeflow; filtered by query param"
-      description: "bindings will contain it's current status"
-      operationId: "read binding"
+    post:
+      summary: "Create a binding specifying a user and an namespace, the binding will\
+        \ grant user edit-access to referred namespace"
+      description: "Only admin or referred namespace owner can create binding"
```
Comment (avdaredevil) [inline components/access-management/api/swagger.yaml]:
```suggestion
      description: "Only admin or referred namespace owner can create a binding"
```

File: components/access-management/api/swagger.yaml [hunk 5]
```diff
@@ -183,37 +208,74 @@ definitions:
...
+        name: "name"
+        namespace: "namespace"
+  Profile_status:
+    properties:
+      status:
+        type: "string"
+        description: "Status of the profile, one of Succeeded, Failed, Unknown."
+      message:
+        type: "string"
+        description: "Non empty when at Failed status, containing detailed error messages"
```
Comment (avdaredevil) [inline components/access-management/api/swagger.yaml]:
```suggestion
        description: "Non-empty when at Failed status, containing detailed error messages"
```

File: components/access-management/kfam/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+This server was generated by the [swagger-codegen]
+(https://github.com/swagger-api/swagger-codegen) project.  
+By using the [OpenAPI-Spec](https://github.com/OAI/OpenAPI-Specification) from a remote server, you can easily generate a server stub.  
+-
+
+To see how to make this your own, look here:
+
+[README](https://github.com/swagger-api/swagger-codegen/blob/master/README.md)
+
+- API version: 1.0.0
```
Comment (lluunn) [inline components/access-management/kfam/README.md]: should we start with 0.1.0?

File: components/access-management/kfam/profiles.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
+package kfam
```
Comment (lluunn) [inline components/access-management/kfam/profiles.go]: seems only this file is not auto-generated?
Other files have the headers.

**Comment by kunmingg** [comment]
Now the only missing logic is auth check before actions.

**Comment by kunmingg** [comment]
WIP removed.

**Comment by kunmingg** [comment]
@jlewi 
e2e test: https://github.com/kubeflow/kubeflow/issues/3458
kustomize: https://github.com/kubeflow/manifests/issues/130

**Pull request merged**